src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+            StackFrameToken closeToken = default;
+
+            while (currentIdentifier.HasValue && currentIdentifier.Value.Kind == StackFrameKind.IdentifierToken)
+            {
+                builder.Add(new StackFrameIdentifierNameNode(currentIdentifier.Value));
+
+                if (_lexer.ScanCurrentCharAsTokenIfMatch(closeBrackeKind, out closeToken))
+                {
+                    break;
+                }
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, out var commaToken))
+                {
+                    return ParseResult<StackFrameTypeArgumentList>.Abort;
+                }
+
+                builder.Add(commaToken);
+                currentIdentifier = _lexer.TryScanIdentifier();
+            }
+
+            if (builder.Count == 0)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            if (closeToken.IsMissing)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            var separatedList = new EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameIdentifierNameNode>(builder.ToImmutable());
+            return new(new(openToken, separatedList, closeToken));
+        }
+
+        /// <summary>
+        /// MyNamespace.MyClass.MyMethod[|(string s1, string s2, int i1)|]
+        /// Takes parameter declarations from method text and parses them into a <see cref="StackFrameParameterList"/>.
+        /// </summary>
+        private ParseResult<StackFrameParameterList> ParseMethodParameters()
+        {
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.OpenParenToken, scanTrailingWhitespace: true, out var openParen))
+            {
+                return ParseResult<StackFrameParameterList>.Abort;
+            }
+
+            if (_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseParenToken, out var closeParen))
+            {
+                return new(new(openParen, EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameParameterDeclarationNode>.Empty, closeParen));
+            }
+
+            using var _ = ArrayBuilder<StackFrameNodeOrToken>.GetInstance(out var builder);
+
+            while (true)
+            {
+                var (success, parameterNode) = ParseParameterNode();
+                if (!success)
+                {
+                    return ParseResult<StackFrameParameterList>.Abort;
+                }
+
+                RoslynDebug.AssertNotNull(parameterNode);
+                builder.Add(parameterNode);
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, out var commaToken))
+                {
+                    break;
+                }
+
+                builder.Add(commaToken);
+            }
+
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseParenToken, out closeParen))
```

Comment:
test with trailing whitespace after this plz.

Reply:
This is covered already in multiple tests